src/middlewares/condom.rs

```diff
@@ -0,0 +1,1305 @@
+                i
+            ));
+        }
+        wasm_code.push('\n');
+
+        // passive data
+        wasm_code.push_str(";; passive data\n");
+        for i in 0..nb_passive_data {
+            wasm_code.push_str(&format!("(data $d{} \"Hello, World!\")\n", i));
+        }
+        wasm_code.push('\n');
+
+        // global initializers
+        wasm_code.push_str(";; global initializers\n");
+        for i in 0..nb_global_ini {
+            wasm_code.push_str(&format!("(global $g{} i32 (i32.const 42))\n", i,));
+        }
+        wasm_code.push('\n');
+
+        // table
+        wasm_code.push_str(";; table\n");
+        for _ in 0..nb_tables {
+            wasm_code.push_str("  (table 1 funcref)\n");
+        }
+        wasm_code.push('\n');
+
+        // decl fn
+        wasm_code.push_str(";; declare functions\n");
+        for i in 0..nb_fn {
+            wasm_code.push_str(&format!("  (func ${} (param", fn_names.get(i).unwrap()));
+
+            for _ in 0..nb_params {
+                wasm_code.push_str(" i32");
+            }
+            wasm_code.push_str(") (result");
+
+            for _ in 0..nb_return_values {
+                wasm_code.push_str(" i32");
+            }
+            wasm_code.push_str(")\n");
+
+            for _ in 0..nb_locals {
+                wasm_code.push_str("    (local i32)\n");
+            }
+
+            wasm_code.push_str("    (block (result");
+            for _ in 0..nb_return_values {
+                wasm_code.push_str(" i32");
+            }
+            wasm_code.push_str(")\n");
+
+            for _ in 0..nb_return_values {
+                wasm_code.push_str(&format!("      i32.const {}\n", i));
+            }
+
+            wasm_code.push_str("    )\n  )\n");
+        }
+        wasm_code.push('\n');
+
+        // export fn
+        wasm_code.push_str(";; export functions\n");
+        for i in 0..nb_exports_fn {
+            wasm_code.push_str(&format!(
+                "  (export \"fn_{}\" (func ${}))\n",
+                i,
+                fn_names.get(i).unwrap()
+            ));
+        }
+        wasm_code.push('\n');
+
+        // table initializer
+        wasm_code.push_str(";; table initializers\n");
+        wasm_code.push_str(&format!("  (table {} funcref)\n", nb_table_ini));
+        for i in 0..nb_table_ini {
+            wasm_code.push_str(&format!(
+                "    (elem (i32.const {}) ${})\n",
+                i,
+                fn_names.get(i).unwrap()
+            ));
+        }
+        wasm_code.push('\n');
+
+        // passive elements
+        wasm_code.push_str(";; passive elements\n");
+        for i in 0..nb_passive_elm {
+            wasm_code.push_str(&format!("  (elem func ${})\n", fn_names.get(i).unwrap()));
+        }
+
+        wasm_code.push(')');
+
+        std::fs::write("calibrate.wat", &wasm_code).unwrap();
+
+        wat2wasm(wasm_code.as_bytes()).unwrap().into()
+    }
+
+    #[test]
+    fn test_condom_middleware_calibrate() {
```

Comment:
@damip @Leo-Besancon @sydhds 
with these limits the memory used to compile the wasm seems to stay around 1.5GB
but on my machine it takes on average 35s.
Please share your mind about this.

Reply:
Updated values for a more realistic compilation time.